src/reqwest/src/async/sniffed_nodes.rs

```diff
@@ -0,0 +1,110 @@
+/*! Connection sniffing and multiple static nodes. */
+
+use std::rc::Rc;
+use std::cell::RefCell;
+use std::sync::Arc;
+use reqwest::unstable::async::Client;
+use futures::{Future, IntoFuture};
+use ::{Error, RequestParams};
+use async::{AsyncElasticClient, AsyncFromResponse};
+use async::nodes_info::NodesInfoResponse;
+use static_nodes::StaticNodes;
+use req::NodesInfoRequest;
+use res::parsing::parse;
+
+/** 
+Periodically sniff nodes in a cluster.
+
+Requests are load balanced between the sniffed nodes using a round-robin strategy.
+The base url for the node is obtained by the `http.publish_address` field on a [node info request].
+*/
+#[derive(Clone)]
+pub struct SniffedNodes {
```

Comment:
Maybe we should introduce some traits to legitimise this pattern a bit better, something like `AsyncGetRequestParams` or whatever that returns a future of some request parameters. Then we've got a clear distinction between `RequestParams` and types that can produce `RequestParams`.